identity-server/src/IdentityServer/Validation/Default/TokenValidator.cs

```diff
@@ -197,18 +198,18 @@ public async Task<TokenValidationResult> ValidateAccessTokenAsync(string token,
         }
 
         // make sure user is still active (if sub claim is present)
-        var subClaim = result.Claims.FirstOrDefault(c => c.Type == JwtClaimTypes.Subject);
+        var subClaim = result.Claims?.FirstOrDefault(c => c.Type == JwtClaimTypes.Subject);
         if (subClaim != null)
         {
-            var principal = Principal.Create("tokenvalidator", result.Claims.ToArray());
+            var principal = Principal.Create("tokenvalidator", result.Claims!.ToArray());
 
             if (result.ReferenceTokenId.IsPresent())
             {
                 principal.Identities.First()
                     .AddClaim(new Claim(JwtClaimTypes.ReferenceTokenId, result.ReferenceTokenId));
             }
 
-            var isActiveCtx = new IsActiveContext(principal, result.Client,
+            var isActiveCtx = new IsActiveContext(principal, result.Client!,
```

Comment:
I don't think we need this !, because the IsActiveContext ctor is going to check for nulls.